qiskit_bot/release_process.py

```diff
@@ -230,9 +230,19 @@ def _get_log_string(version_obj):
             f"{version_obj.micro - 1}"
         )
     # If a minor release log between 0.X.0..0.X-1.0
+    elif version_obj.major >= 1 and version_obj.minor == 0:
+        tags = git.get_tags(repo)
+        previous_major = version_obj.major - 1
+        for tag in tags.splitlines():
+            tag_version = parse(tag)
+            if tag_version.is_prerelease:
+                continue
+            if tag_version.major == previous_major:
+                old_version = tag
+                break
     else:
         old_version = f"{version_obj.epoch}.{version_obj.minor - 1}.0"
```

Comment:
The comment at the top here looks like it's now attached to the wrong branch of this `if/elif` chain now.

Also, this might be a bit tricksy with regard to the Qiskit 0.46 strategy - the changelog we'll actually care about will typically be the penultimate minor, in the "final minor is the deprecation boogaloo" style we were planning to do with our majors, I think?  With things as currently written, and how we've been merging up from 0.45 to 0.46, I think the `...` comparison is going to end up including the bugfixes from 0.45.1+ in the changelog for 1.0?  Not certain about that.

Reply:
I had forgotten about the 0.46.0 release when I wrote this logic. If we tag 0.46.0 before we tag 1.0.0rc1 then this will be `1.0.0rc1...0.46.0`, while if we tag rc1 before 0.46.0 it will be `1.0.0rc1...0.45.x`. At the end of the day I don't think it'll make a huge difference as long as we forward port any bugfixes from `stable/0.45` to `stable/0.46` it will be basically the same history I think.